### PR TITLE
Soumyo/ec2 transit gateway route table

### PR DESCRIPTION
**Pull request by soumyo13**
### Description

`ec2 transit gateway route table` InSpec audit resource to test properties of a ec2 transit gateway route table.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/integration/build/aws.tf [hunk 1]
```diff
@@ -1941,4 +1941,8 @@ resource "aws_elasticache_replication_group" "replication_group" {
   node_type                     = var.aws_elasticache_replication_group_node_type
   at_rest_encryption_enabled    = true
   transit_encryption_enabled    = false
+}
+
+resource "aws_ec2_transit_gateway_route_table" "aws_ec2_transit_gateway_route_table1" {
+  transit_gateway_id = "tgw-02850dffe1c3b222c"
```
Comment (sathish-progress) [inline test/integration/build/aws.tf:1947]: this should not be hardcoded! ..

File: test/integration/verify/controls/aws_ec2_transit_gateway_route_table.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+aws_transit_gateway_route_table_id = attribute(:aws_transit_gateway_route_table_id, default: '', description: 'The ID of the transit gateway route table.')
+aws_transit_gateway_id = attribute(:aws_transit_gateway_id, default: '', description: 'The ID of the transit gateway.')
+aws_state = attribute(:aws_state, default: '', description: 'The state of the route table (available | deleting | deleted | pending).')
+aws_default_association_route_table = attribute(:aws_default_association_route_table, default: '', description: 'Indicates whether this is the default association route table for the transit gateway (true | false).')
+aws_default_propagation_route_table = attribute(:aws_default_propagation_route_table, default: '', description: 'Indicates whether this is the default propagation route table for the transit gateway (true | false)')
+aws_creation_time = attribute(:aws_creation_time, default: '', description: 'The creation time of the transit gateway route table.')
+aws_tags = attribute(:aws_tags, default: '', description: 'The tags of the transit gateway route table.')
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_ec2_transit_gateway_route_table.rb:7]: where are these attrs?

**Comment by sathish-progress** [comment]
please update resource doc links to readme

File: test/integration/build/aws.tf (hunk 1 above)
Comment (soumyo13) [reply test/integration/build/aws.tf:1947]: done

**Comment by soumyo13** [comment]
> please update resource doc links to readme

done

File: test/integration/verify/controls/aws_ec2_transit_gateway_route_table.rb (hunk 2 above)
Comment (soumyo13) [reply test/integration/verify/controls/aws_ec2_transit_gateway_route_table.rb:7]: done

**Review by Nirbhay1997 — APPROVED** [review]
LGTM

**Pull request merged**